python/ml4ir/base/model/relevance_model.py

```diff
@@ -778,3 +779,12 @@ def _build_callback_hooks(
         # Add more here
 
         return callbacks_list
+
+    def calibrate(self, relevance_dataset, logger, logs_dir_local, temperature_init):
```

Comment:
I am a bit confused now. I see the `NotImplemented` above in ranking model. Should it come here, so that the models inherit it (so we do not touch the ranking file) and define this `calibrate` on the `ml4ir/applications/classification/model/classification_model.py`? Is this implementation specific to classification approaches or classification data?

Reply:
It is expected to have calibration for ranking in future. @lastmansleeping  also recommended to keep `calibration` here.

Reply:
This is mostly in hopes that we can have something generic that we can ship with `RelevanceModel`. But if we realize that it doesn't work, we can move it. I like the idea of having a common base implementation and overrides if necessary.